test/e2e/snaps/test-snap-txinsights.spec.js

```diff
@@ -0,0 +1,116 @@
+        await driver.press('#password', driver.Key.ENTER);
+
+        // navigate to test snaps page and connect
+        await driver.driver.get(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+
+        // find and scroll to the bip32 test and connect
+        const snapButton1 = await driver.findElement('#connectInsightsSnap');
+        await driver.scrollToElement(snapButton1);
+        await driver.delay(1000);
+        await driver.clickElement('#connectInsightsSnap');
+
+        // switch to metamask extension and click connect
+        let windowHandles = await driver.waitUntilXWindowHandles(
+          2,
+          1000,
+          10000,
+        );
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement(
+          {
+            text: 'Connect',
+            tag: 'button',
+          },
+          10000,
+        );
+
+        await driver.delay(2000);
+
+        // switch to metamask extension
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+
+        // approve install of snap
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          text: 'Approve & install',
+          tag: 'button',
+        });
+
+        await driver.delay(1000);
+
+        // switch to test-snaps page and get accounts
+        windowHandles = await driver.waitUntilXWindowHandles(1, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+        await driver.clickElement('#getAccounts');
+        await driver.delay(1000);
+
+        // switch back to MetaMask window and switch to tx insights pane
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({
+          text: 'Next',
+          tag: 'button',
+        });
+        await driver.delay(1000);
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        // switch to test-snaps page and send tx
+        windowHandles = await driver.waitUntilXWindowHandles(1, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+        await driver.clickElement('#sendInsights');
+        await driver.delay(1000);
+
+        // switch back to MetaMask window and switch to tx insights pane
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.delay(2000);
+        await driver.clickElement({
+          text: 'TxInsightsTest',
+          tag: 'button',
+        });
```

Comment:
This ask is waiting for the following PR: https://github.com/MetaMask/metamask-extension/pull/16814
The plan was to release a patch to the test once it was merged. 
The thought was that the `TXINSIGHTSTEST` tab showing up and being clickable was a good start to a test to prove that a snap was loaded with that permission and it works, since this is not currently tested.

Reply:
This is fixed with [91622ce](https://github.com/MetaMask/metamask-extension/pull/16785/commits/91622cee46fc22012ca5e277b7d54496fbf5ec1d)